common/ECSResourcesCheck/src/main/java/ECSResources.java

```diff
@@ -0,0 +1,111 @@
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.EnvironmentVariableCredentialsProvider;
+import com.amazonaws.services.ecs.AmazonECS;
+import com.amazonaws.services.ecs.AmazonECSClientBuilder;
+import com.amazonaws.services.ecs.model.DescribeServicesRequest;
+import com.amazonaws.services.ecs.model.DescribeServicesResult;
+import com.amazonaws.services.ecs.model.DescribeTaskDefinitionRequest;
+import com.amazonaws.services.ecs.model.DescribeTaskDefinitionResult;
+import org.json.JSONObject;
+
+public class ECSResources {
+    private AWSCredentialsProvider _credentialsProvider;
+    private AmazonECS _ecsClient;
+    private String _cluster;
+
+    public ECSResources (String cluster,String aws_access_key_id,String aws_secret_access_key, String region){
+        AWSCredentialsProvider credentialsProvider=new AWSStaticCredentialsProvider((new BasicAWSCredentials(aws_access_key_id, aws_secret_access_key)));
+        _ecsClient= AmazonECSClientBuilder
+                .standard()
+                .withCredentials(credentialsProvider)
+                .withRegion(region)
+                .build();
+        _cluster=cluster;
+
+    }
+
+    public ECSResources(AmazonECS ecsClient,String cluster){
+        _ecsClient = ecsClient;
+        _cluster=cluster;
+
+    }
+
+
+    public String describeService (String Service){
+        DescribeServicesRequest describeServicesRequest=new DescribeServicesRequest();
+        describeServicesRequest.setCluster(_cluster);
+        describeServicesRequest.withServices(Service);
+        DescribeServicesResult describeServicesResult =_ecsClient.describeServices(describeServicesRequest);
+        if (describeServicesResult.getFailures().size()>0){
+            return null;
+        }
+        return  describeServicesResult.getServices().get(0).getTaskDefinition();
+    }
+
+    public int getDesiredCount (String Service){

```

Comment:
I don't think this method or the ones starting here to the rest of the page should be in this ECSResources.java class.  I think they are helper functions to retrieve a property of the task def. Either you could create a static helper Task Def class that has these static methods or put these methods inside Comparer.java.

Reply:
added new class